src/index.ts

```diff
@@ -1,168 +1,43 @@
-};
-
-const handleLinkedIDEvent = (event: LinkedIDEvent): LinkedIDEvent => {
-  console.log(event.ph);
-  return event;
-};
-
-events.forEach((event: Event): void => {
-  switch (event.ph) {
-    case EventsPhase.METADATA_EVENTS:
-      handleMetadataEvent(event);
-      break;
-
-    case EventsPhase.DURATION_EVENTS_BEGIN:
-    case EventsPhase.DURATION_EVENTS_END:
-      handleDurationEvent(event);
-      break;
-
-    case EventsPhase.COMPLETE_EVENTS:
-      handleCompleteEvent(event);
-      break;
-
-    case EventsPhase.SAMPLE_EVENTS:
-      handleSampleEvent(event);
-      break;
-
-    case EventsPhase.OBJECT_EVENTS_CREATED:
-    case EventsPhase.OBJECT_EVENTS_SNAPSHOT:
-    case EventsPhase.OBJECT_EVENTS_DESTROYED:
-      handleObjectEvent(event);
-      break;
-
-    case EventsPhase.CLOCK_SYNC_EVENTS:
-      handleClockSyncEvent(event);
-      break;
-
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_START:
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_INSTANT:
-    case EventsPhase.ASYNC_EVENTS_NESTABLE_END:
-      handleAsyncEvent(event);
-      break;
-
-    case EventsPhase.CONTEXT_EVENTS_ENTER:
-    case EventsPhase.CONTEXT_EVENTS_LEAVE:
-      handleContextEvent(event);
-      break;
-    case EventsPhase.INSTANT_EVENTS:
-      handleInstantEvent(event);
-      break;
-
-    case EventsPhase.COUNTER_EVENTS:
-      handleCounterEvent(event);
-      break;
-
-    case EventsPhase.FLOW_EVENTS_START:
-    case EventsPhase.FLOW_EVENTS_STEP:
-    case EventsPhase.FLOW_EVENTS_END:
-      handleFlowEvent(event);
-      break;
-
-    case EventsPhase.MEMORY_DUMP_EVENTS_GLOBAL:
-    case EventsPhase.MEMORY_DUMP_EVENTS_PROCESS:
-      handleMemoryDumpEvent(event);
-      break;
-
-    case EventsPhase.MARK_EVENTS:
-      handleMarkEvent(event);
-      break;
-
-    case EventsPhase.LINKED_ID_EVENTS:
-      handleLinkedIDEvent(event);
-      break;
-    default: {
-      // This case should only be hit if we've forgotten to handle
-      // a type of event, or if the incoming data does not match any
-      // of the expected phases at runtime
-      throw new Error(
-        `Unknown event type "${
-          (event as any).ph
-        }" in trace event ${JSON.stringify(event)}`
-      );
-    }
+import { readFileSync, writeFileSync } from 'fs';
+import ip from 'ip';
+import minimist from 'minimist';
+
+import { CpuProfilerModel } from './profiler/cpuProfilerModel';
+import { DurationEvent } from 'types/EventInterfaces';
+
+const argv = minimist(process.argv.slice(2));
+if (!('port' in argv)) {
+  console.log('No DEBUG_PORT provided, using 8081 by default');
+} else {
+  console.log(
+    `Make sure the React Native Debugging server is running on port ${argv['port']} on your local machine`
+  );
+}
+
+/* Read Hermes Profile
+ **Replace with CLI transform function as it will just give the path of the CPU profile,
+ **adopting same model here
+ */
+
+const PROFILE_PATH = 'nestedFuncProfile.cpuprofile';
+const DEBUG_SERVER_PORT: string =
```

Comment:
The responsibility of the `hermes-profile-transformer` package should be to export a single function that's able to take in a profile file path and optionally a source map file path, run the transformation, and provide an output as a string. It should not be aware of React Native.

The responsibility of the react-native cli is to provide the inputs.

It's ok for this package to also provide a minimal standalone CLI to run the transformation (when not being used from react-native cli, but that should be in a separate file and provided in the [npm `bin` field](https://docs.npmjs.com/files/package.json#bin). The index file should be just the exported code.